### PR TITLE
Phase 6: Ensemble Seeds 66-73 (Building Toward 24-Model)

### PR DESCRIPTION
**Pull request by tcapelle**
Train 8 more seeds for ensemble expansion. After training: combine with existing 42-57 seeds for 24-model ensemble evaluation.

ALL 8 GPUs: seeds 66-73.

```bash
for seed in 66 67 68 69 70 71 72 73; do
  python train.py --agent tanjiro --wandb_group phase6/ensemble-more-seeds \
    --asinh_pressure --asinh_scale 0.75 + all baseline flags --seed ${seed}
done
```

Current 8-ensemble: p_in 12.4, p_oodc 6.7, p_tan 29.4, p_re 5.8

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: All 8 runs finished — great work! Individual seed metrics look consistent with the 8-seed baseline (p_in=13.06±0.34, p_oodc=7.88±0.17, p_tan=30.29±0.63, p_re=6.42±0.09).

**Next step:** Please evaluate the 16-seed ensemble by combining your seeds (66-73) with the original 8 seeds (42-49). The original 8 W&B run IDs are: rboyvjeo, h0uog211, kwt8tw52, 5j26p5v1, rmump7ke, ujt9cu0l, 7fw8ksxq, 0lsry8km.

Run eval_ensemble.py with all 16 run IDs and report the 16-model ensemble surface MAE (p_in, p_oodc, p_tan, p_re). Compare to the current 8-seed ensemble baseline (p_in=12.4, p_oodc=6.7, p_tan=29.4, p_re=5.8).

This is the key test: does 16-seed beat 8-seed?

**Comment by tcapelle** [comment]
STUDENT tanjiro:

## Results

### Individual Model Metrics (Seeds 66-73)

| Seed | Run ID | Epochs | p_in | p_oodc | p_tan | p_re | val/loss |
|------|--------|--------|------|--------|-------|------|----------|
| 66 | j9w7d1r7 | 155 | 13.0 | 7.6 | 30.4 | 6.2 | 0.3872 |
| 67 | mc4jvgqj | 156 | 12.9 | 7.9 | 30.6 | 6.3 | 0.3874 |
| 68 | cbbvhl62 | 155 | 12.9 | 8.0 | 29.9 | 6.5 | 0.3875 |
| 69 | bigqfn3k | 154 | 13.0 | 7.9 | 30.1 | 6.4 | 0.3899 |
| 70 | bqhg6lq8 | 155 | 12.7 | 8.1 | 31.5 | 6.5 | 0.3926 |
| 71 | 5ukk7wv6 | 155 | 13.7 | 7.9 | 30.4 | 6.5 | 0.3897 |
| 72 | xlnhwuqc | 155 | 13.4 | 8.0 | 30.0 | 6.5 | 0.3893 |
| 73 | ii1tz4vv | 154 | 12.8 | 7.7 | 29.4 | 6.4 | 0.3830 |
| **Mean** | | **154.9** | **13.1** | **7.9** | **30.3** | **6.4** | **0.3883** |
| **Std** | | | **0.3** | **0.2** | **0.6** | **0.1** | **0.003** |

### 8-Seed Ensemble (Seeds 66-73)

| Metric | New 8-ensemble (66-73) | Existing 8-ensemble (42-49) | Delta |
|--------|------------------------|----------------------------|-------|
| p_in | **12.2** | 12.4 | **-1.6%** |
| p_oodc | **6.7** | 6.7 | 0% |
| p_tan | **29.1** | 29.4 | **-1.0%** |
| p_re | **5.8** | 5.8 | 0% |

### Training Details

- **Command per seed:**
  ```bash
  CUDA_VISIBLE_DEVICES=<gpu> python train.py --agent tanjiro \
    --wandb_name "tanjiro/ensemble-seed-<seed>" --seed <seed> \
    --wandb_group phase6/ensemble-more-seeds \
    --field_decoder --adaln_output --use_lion --lr 2e-4 \
    --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
    --n_layers 3 --slice_num 96 --tandem_ramp \
    --domain_layernorm --domain_velhead --ema_decay 0.999 \
    --weight_decay 5e-5 --cosine_T_max 160 --disable_pcgrad \
    --pressure_first --pressure_deep --residual_prediction \
    --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
    --asinh_pressure --asinh_scale 0.75
  ```
- **Peak memory:** 38.0 GB per model
- **Training time:** ~180 min per model (hit wall-clock limit at epoch 154-156 of 160 target)
- **W&B group:** `phase6/ensemble-more-seeds`
- **Run IDs:** j9w7d1r7, mc4jvgqj, cbbvhl62, bigqfn3k, bqhg6lq8, 5ukk7wv6, xlnhwuqc, ii1tz4vv

### What Happened

All 8 seeds trained successfully on 8 GPUs in parallel. Training hit the 180-minute wall-clock timeout at epoch 154-156 (target cosine T_max=160), so models are ~97% through the LR schedule. This should have minimal impact on final quality.

The new 8-seed ensemble (66-73) matches or slightly improves on the existing 8-ensemble (42-49): p_in improved by 1.6%, p_tan by 1.0%, while p_oodc and p_re are unchanged. Individual model variance is consistent with prior characterization (std ~0.3 for p_in, ~0.2 for p_oodc).

**Note:** Checkpoints for seeds 42-57 are not available on this node, so I could not run the full 24-model ensemble evaluation. The advisor will need to combine all run IDs for the combined evaluation.

### For 24-Model Ensemble Evaluation

New seeds (66-73): `j9w7d1r7 mc4jvgqj cbbvhl62 bigqfn3k bqhg6lq8 5ukk7wv6 xlnhwuqc ii1tz4vv`
Existing seeds (42-49): `rboyvjeo h0uog211 kwt8tw52 5j26p5v1 rmump7ke ujt9cu0l 7fw8ksxq 0lsry8km`

### Suggested Follow-ups

- Run the full 24-model ensemble evaluation once all seed checkpoints (42-57, 66-73) are co-located
- Seeds 50-57 run IDs needed for the complete 24-model ensemble
- Consider whether the 5-6 missing epochs (reached 154-156 vs target 160) warrant extending the timeout slightly for future runs

**Comment by tcapelle** [comment]
ADVISOR: Merging — new 8-seed ensemble (seeds 66-73) beats the current baseline on p_in (-1.6%) and p_tan (-1.0%).

New best: p_in=**12.2**, p_oodc=6.7, p_tan=**29.1**, p_re=5.8

W&B run IDs: j9w7d1r7 mc4jvgqj cbbvhl62 bigqfn3k bqhg6lq8 5ukk7wv6 xlnhwuqc ii1tz4vv

Next step: combine with seeds 42-49 for the full 16-seed ensemble evaluation — the combined ensemble should push these metrics even lower. Will handle this in the next round.
